### ISSUE TITLE
EFS mount helper doesn't support profiles with assume role

### ISSUE THREAD

**Issue by jsharpe** [aws/efs-utils#77]
EFS mount helper should support assumed roles via profiles e.g.

[myrole]
credential_source = Ec2InstanceMetadata
role_arn = arn:aws:iam::1234567890:role/myrole

**Comment by Cappuccinuo** [comment]
Thanks for the feedback, we'll have someone look into this.

**Comment by clintoncampbell** [comment]
Just ran into this as well, though I think it might be better to call this a bug rather than a feature request based on the current state of the documentation. From what I can tell from testing the awsprofile is failing in all but a very narrow range of use cases. 

For reference, I am trying to mount the filesystem from offsite with an IAM identity and profile (for management purposes). I am working from an Amazon Linux 2 docker image. I have version 1.26.3 of efs-utils installed. The mount operation looks like:

```
mount -v -t efs -o tls,iam,awsprofile=myprofile fs-xxxxxx /mnt/efs
```

As described in EFS documentation, I've added an entry to /etc/host to associate the mount point IP with the derived IP hostname of this filesystem.

I ran into trouble first with the helper detecting named config profiles. After some experimentation, I discovered that it does not properly detect the standard [profile myprofile] within config. Instead, I needed to duplicate my profile in .aws/credentials as [myprofile] and replace the source_profile directive with a duplicate copy of the credentials. 

After resolving the previous issues, I finally received the `mount.nfs4: access denied by server while mounting 127.0.0.1:/` error while connecting. Looking at the NewClientConnection event in CloudTrail, I discovered that the helper had not passed any IAM credentials -- neither the base identity or the assumed role -- and was connecting as an anonymous user in the account.

Thanks to all of the contributors for your work on this. Please let me know if I can provide additional detail.

**Comment by Cappuccinuo** [comment]
Fix in v1.31.1, please follow [this part](https://github.com/aws/efs-utils#use-the-assumed-profile-credentials-for-iam) to use the assumed awsprofile.
